main.js

```diff
@@ -52,60 +52,114 @@ const DEFAULT_PAGE_SIZE = 64; // Page size for retrieving documents from the cus
 
   const options = program.opts();
 
-  if (options.validate) {
-    await validate(options);
-  } else {
+  const migrator = new MigrationClient({
+    sourceKey: options.source,
+    targetKey: options.target,
+    defaultPageSize: DEFAULT_PAGE_SIZE,
+    maxParallelism: options.parallelism,
+  });
+
+  const collectionList = [];
+  if (options.collection) {
     const collectionName = options.collection;
     const indexName = options.index ?? "_migration_index_for_" + collectionName;
-
-    const migrator = new MigrationClient({
-      sourceKey: options.source,
-      targetKey: options.target,
-      defaultPageSize: DEFAULT_PAGE_SIZE,
-      maxParallelism: options.parallelism,
+    collectionList.push({
+      collectionName,
+      indexName,
     });
+  } else {
+    if (options.index) {
+      console.log(
+        "Custom index specified, but cannot be used when running with multiple collections.",
+      );
```

Comment:
Need to exit here?

Reply:
Okay. That would let the user handle this more intentionally.

Reply:
Oh, I see; this is just a warning that the index passed to the options is going to be ignored in this case.  Maybe tweak the message to that effect.

Reply:
Yeah, exiting probably makes sense.  Honestly at this point, having a custom index name might not be necessary; if we document what the index name is, they could conceivably create it in advance if they really wanted to with the correct name.  Thoughts?

Reply:
But if they just happen to have the right index shape on a very large collection but it doesn't have the right name, that would be frustrating to have to wait for the redundant index build.  🤷

Reply:
Updating the message to explain why it's going to fail and then fail sounds good.

perhaps since we will update to provide a list of collection names, we can also accept a list of index names. Then we can validate that the length of both lists are the same.

In the case we fallback to search for and list all collections, then we should can still fail if index names are provided.

How's that?